### PR TITLE
Fix rollback on sle15

### PR DESCRIPTION
**Pull request by felixsch**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/106531/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9851c9e60e1705fc90735313442c2ad876d37321 into b104e146312a51dd2b4e2bd27702cd885e3b45b4)

File: spec/connect/migration_spec.rb [hunk 1]
```diff
@@ -17,33 +17,42 @@
     let(:config) { SUSE::Connect::Config.new }
     let(:client) { SUSE::Connect::Client.new(config) }
     let(:status) { SUSE::Connect::Status.new(config) }
+    let(:product_tree) { JSON.parse(File.read('spec/fixtures/product_tree_migration.json')) }
+    let(:product) { Remote::Product.new(product_tree) }
+
     let(:service) do
-      SUSE::Connect::Remote::Service.new(name: 'SLES12', obsoleted_service_name: 'SLES11',
```
Comment (ikapelyukhin) [inline spec/connect/migration_spec.rb]: Can we keep the SLE12 specs? We need it working on both, I would be good to have both scenarios covered with tests.

Comment (felixsch) [reply spec/connect/migration_spec.rb]: Do you mean I should create two tests, yes for sure if you think that's the way to go :)

File: lib/suse/connect/migration.rb [hunk 2]
```diff
@@ -23,9 +23,14 @@ def rollback(client_params = {})
           client = Client.new(config)
           status = Status.new(config)
 
-          # FIXME: Sort products and ensure the base product is the first one in the list
-          status.installed_products.sort_by { |p| p.isbase ? 0 : 1 }.each do |product|
-            service = client.downgrade_product(product)
+          tree = client.show_product(Zypper.base_product)
```
Comment (felixsch) [inline lib/suse/connect/migration.rb]: This calls the product endpoints but it doesn't work if no system is known after the rollback. Find a better solution!

Comment (ikapelyukhin) [reply lib/suse/connect/migration.rb]: :+1:

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/107917/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 93c9b8cf4ca952337afc6d9814351a1c41843d9b into 2b4a9463c577c0b717675b772f44ae9527f0d5c2)

File: spec/fixtures/product_tree_migration_sle12.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "identifier": "SLES",
+  "version": 12,
+  "recommended": false,
+  "arch": "x86_64",
+  "isbase": true,
+  "extensions": [
+    {
+      "identifier": "sle-module-legacy",
+      "version": 12,
+      "recommended": true,
```
Comment (ikapelyukhin) [inline spec/fixtures/product_tree_migration_sle12.json]: Hmm, for a SLES12 module?

File: lib/suse/connect/migration.rb [hunk 4]
```diff
@@ -90,6 +97,18 @@ def find_products(identifier)
         def install_release_package(identifier)
           SUSE::Connect::Zypper.install_release_package(identifier)
         end
+
+        # Removes service an readds service with current url and name
```
Comment (ikapelyukhin) [inline lib/suse/connect/migration.rb]: `an readds` -> `and readds`

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/107932/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 8771a7cc31a62b097dba6b2e06106f2f9eb36d5b into 2b4a9463c577c0b717675b772f44ae9527f0d5c2)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/107948/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0ec9c017975cebce5fa388906dfc749fde1cacbd into 2b4a9463c577c0b717675b772f44ae9527f0d5c2)

**Pull request merged**